apps/builder/app/builder/features/ai/operations.tsx

```diff
@@ -0,0 +1,203 @@
+              signal: abort.current.signal,
+            },
+          ],
+          { retry: 2 }
+        )
+          .then((result) => {
+            if (result.success === false) {
+              setError(result.message);
+              return;
+            }
+            applyOperations(result.data);
+          })
+          .finally(() => {
+            abort.current = undefined;
+            setIsLoading(false);
+          });
+
+        setError(undefined);
+        setIsLoading(true);
+      }}
+    >
+      <Label>
+        Prompt
+        <InputField name="prompt" />
+      </Label>
+      <input type="hidden" name="data" value={JSON.stringify(jsx)} />
+      <Button
+        onClick={(event) => {
+          if (isLoading) {
+            event.preventDefault();
+            abort.current?.abort();
+          }
+        }}
+      >
+        {isLoading ? "stop" : "go"}
+      </Button>{" "}
+      {isLoading ? <SpinnerIcon /> : null}
+      {error ? <Text>{error}</Text> : null}
+    </form>
+  );
+};
+
+const $jsx = computed(
+  [
+    selectedInstanceSelectorStore,
+    instancesStore,
+    propsStore,
+    dataSourcesStore,
+    registeredComponentMetasStore,
+    breakpointsStore,
+    stylesStore,
+    styleSourceSelectionsStore,
+  ],
+  (
+    selectedInstanceSelector,
+    instances,
+    props,
+    dataSources,
+    metas,
+    breakpoints,
+    styles,
+    styleSourceSelections
+  ) => {
+    if (selectedInstanceSelector === undefined) {
+      return null;
+    }
+
+    const [rootInstanceId] = selectedInstanceSelector;
+    const instance = instances.get(rootInstanceId);
+    if (instance == null) {
+      return null;
+    }
+    const indexesWithinAncestors = getIndexesWithinAncestors(metas, instances, [
+      rootInstanceId,
+    ]);
+    const scope = createScope();
+
+    const jsx = generateJsxElement({
+      scope,
+      instance,
+      props,
+      dataSources,
+      indexesWithinAncestors,
+      children: generateJsxChildren({
+        scope,
+        children: instance.children,
+        instances,
+        props,
+        dataSources,
+        indexesWithinAncestors,
+      }),
+    });
+
+    const treeInstanceIds = findTreeInstanceIds(instances, rootInstanceId);
+
+    const treeStyleSourceSelections = new Map(
+      getMapValuesByKeysSet(styleSourceSelections, treeInstanceIds).map(
+        (styleSourceSelection) => [
+          styleSourceSelection.instanceId,
+          styleSourceSelection,
+        ]
+      )
+    );
+
+    const engine = createCssEngine({ name: "ssr" });
+
+    const styleRules = getStyleRules(styles, treeStyleSourceSelections);
+    for (const { breakpointId, instanceId, state, style } of styleRules) {
+      engine.addStyleRule(`[${idAttribute}="${instanceId}"]${state ?? ""}`, {
+        breakpoint: breakpointId,
+        style,
+      });
+    }
+
+    return `<style>{\`${engine.cssText.replace(/\n/gm, " ")}\`}</style>${jsx
+      .replace(/data-ws-component="[^"]+"/g, "")
```

Comment:
we do have a constant for this attribute name, in case we want to change it in the future, search for "componentAttribute", maybe we need a better place than react-sdk for those attributes?

@TrySound